src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -431,6 +431,11 @@ private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnosti
                     Error(diagnostics, ErrorCode.ERR_FieldCantBeRefAny, argumentSyntax, argumentType);
                 }
 
+                if (argumentType.IsPointerType())
+                {
+                    Error(diagnostics, ErrorCode.ERR_BadTypeArgument, argumentSyntax, argumentType);
```

Comment:
In other places (Binder_Expressions.cs and ConstraintsHelper.cs), I see that the same error message is used both for restricted types and pointer types. 
I think it's probably better to do the same here `if (argumentType.IsRestrictedType() || argumentType.IsPointerType()) ...`